### PR TITLE
[RFC] Fix coverity issues. (6)

### PR DESCRIPTION
**Pull request by elmart**
This gets rid of the last 3 issues in "High Impact Outstanding" category, and 1 other one which happened to involve code touched by one of the previous ones.

### CODE REVIEW

**Comment by elmart** [comment]
Review commit added.

**Comment by elmart** [comment]
Repushed last review commit.

**Comment by jszakmeister** [comment]
I'm not sure why there's a difference between my QB setup and Travis, but something is segfaulting under QB and the linux agent. :-(

**Comment by jszakmeister** [comment]
This may have been my fault.  I started the agent as the wrong user.  But it's interesting that we ran into an interaction that caused a segfault. :-(

File: src/nvim/os/pty_process.c [hunk 1]
```diff
@@ -69,6 +70,26 @@ void pty_process_destroy(Job *job)
   job->process = NULL;
 }
 
+static bool set_pipe_duplicating_descriptor(int fd, uv_pipe_t *pipe)
```
Comment (oni-link) [inline src/nvim/os/pty_process.c]: `FUNC_ATTR_NONNULL_ALL` could be added.

Comment (elmart) [inline src/nvim/os/pty_process.c]: :+1: 
Repushed last review commit.

**Comment by elmart** [comment]
@oni-link Is this ok to you?

**Comment by oni-link** [comment]
LGTM.

**Comment by elmart** [comment]
As my proposal at #2122 about leaving review commits authored by reviewers didn't have much success (and I agree it's not ideal, because of log pollution), I've opted for the other suggested alternative: adding Reviewed-by and Helped-by tags to commit messages.

**Pull request merged**